rust/lesson-plan.md

```diff
@@ -0,0 +1,315 @@
+project comes with a test suite, and the project should not be considered
+finished until it passes with `cargo test`.
+
+You should not read the example code for any project until completing the
+project yourself. You are encouraged though to learn from and apply techniques
+they contain to your own projects retroactively. Keep in mind though that the
+example projects are not the only way &mdash; or even the best way &mdash; to
+solve the problems. Trust yourself and your own creativity.
+
+> RE plagiarism (this is mostly relevant to students being evaluated on their
+coursework): The line between applying techniques learned by code-reading and
+copying code outright can be hard to identify. But as a professional you have
+ethical responsibilites, and only you can know if you are upholding them. For
+those not being evaluated for their coursework, simply copying from the example
+code will only hurt your experience; for those who are being evaluated, your
+instructors and evaluators are expecting you to use your own skills.
+
+You will recieve further instruction about setting up the source code and test
+suite, as well as project specifications, as you progress through the individual
+projects.
+
+The expected time to complete each section is not currently well-estimated, but
+the building blocks might take up to 8 hours, and the projects up to 40 hours.
+If you are spending much less time than that, or are spending more time, don't
+worry: these are just bad estimates, and everybody's experience is different.
+
+
+## Getting help
+
+You will run into problems that you don't know how to solve. Don't suffer in
+silence. Everybody needs help sometimes.
+
+And fortunately the Rust community is amazing and welcoming, and wants to help
+you. As you are progressing through this course, you are _strongly_ encouraged
+to join the Rust community experience.
+
+Here are the resources you should consider turning to first:
+
+- The #rust-training channel on the [TiKV Slack]. This channel exists just for
+  this course. Please consider joining to support your fellow students and other
+  course-takers. There will always be someone there to answer questions, but
+  there may be a delay due to time zones and other factors. Both English and
+  Chinese languages are welcome here.
+
+- The #beginners channel on the official [Rust Discord]. You are almost
+  guaranteed to get some answer here, and if not, don't hesitate to ask again.
+  The people who hang out here are there specifically to help. Because of time
+  zone differences it may take time for somebody to respond. Only English will
+  be consistently understood here.
+
+- The QQ Rust group ([QR code][qq]). For Chinese students, this is one of the
+  major Rust communities in China. This is a great place to hang out generally,
+  and for those unconfident in their English skills, a great place to ask
+  for help.
+
+- The WeChat Rust group #2 ([QR code][wechat]). Again this is mostly for the
```

Comment:
The QR code refers to another Rust QQ group II instead of a wechat group...

Reply:
:-/

Reply:
Thank you very much for catching that for me.